hr_utilization_analysis/report/hr_utilization_analysis.py

```diff
@@ -52,18 +52,88 @@ def _compute_entry_ids(self):
         for analysis in self:
             employee_ids = HrEmployee.search(analysis._get_employees_domain())
 
-            entry_ids = [(5, False, False)]
-            for employee_id in employee_ids:
-                date = analysis.date_from
-                one_day = timedelta(days=1)
-                while date <= analysis.date_to:
-                    entry_ids.append(
-                        (0, False, {"employee_id": employee_id.id, "date": date})
-                    )
-                    date += one_day
+            dates = []
+            date = analysis.date_from
+            one_day = timedelta(days=1)
+            while date <= analysis.date_to:
+                dates.append(date)
+                date += one_day
+
+            entry_ids = [(5, False, False)] + [
+                (0, False, d) for d in self._get_entry_id_dicts(employee_ids, dates)
+            ]
 
             analysis.entry_ids = entry_ids
 
+    def _get_entry_id_dicts(self, employees, dates):
+        Module = self.env["ir.module.module"]
+        AccountAnalyticLine = self.env["account.analytic.line"]
+
+        project_timesheet_holidays = Module.with_user(SUPERUSER_ID).search(
+            [("name", "=", "project_timesheet_holidays"), ("state", "=", "installed")],
+            limit=1,
+        )
+
+        uom_hour = self.env.ref("uom.product_uom_hour")
+
+        all_line_ids = AccountAnalyticLine.search(
+            [
+                ("project_id", "!=", False),
+                ("employee_id", "in", employees.ids),
+                ("date", "in", dates),
+            ]
+        )
+        entries = []
+        for employee in employees:
+            tz = pytz.timezone(employee.resource_calendar_id.tz)
+            from_datetime = datetime.combine(min(dates), time.min).replace(tzinfo=tz)
+            to_datetime = datetime.combine(max(dates), time.max).replace(tzinfo=tz)
+            work_time = employee.list_work_time_per_day(from_datetime, to_datetime)
+            hours_dict = {w[0]: w[1] for w in work_time}
+            if project_timesheet_holidays:
+                leaves = employee.list_leaves(from_datetime, to_datetime)
+                leaves_dict = {e[0]: e[1] for e in leaves}
+
+            for date in dates:
+                line_ids = all_line_ids.filtered(
+                    lambda l: l.employee_id == employee and l.date == date
+                )
+                base_entry = {"employee_id": employee.id, "date": date}
```

Comment:
I'd extract this into a separate method then for extensibility

Reply:
Which method would you extract?

Reply:
`_get_entry_base_values()`?